test/test_linalg.py

```diff
@@ -4028,21 +4028,35 @@ def test_orgqr_errors_and_warnings(self, device):
         with self.assertRaisesRegex(RuntimeError, "tau dtype Int does not match input dtype"):
             torch.orgqr(reflectors, tau.to(torch.int))
 
-        # TODO: enable the following tests when orgqr is implemented for CUDA
         if torch.cuda.is_available():
-            with self.assertRaisesRegex(RuntimeError, "the operator doesn't exist for this backend"):
-                # device of out and input should match
-                wrong_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
-                out = torch.empty_like(reflectors).to(wrong_device)
-                # with self.assertRaisesRegex(RuntimeError, "Expected result and input to be on the same device"):
+            # device of out and input should match
+            wrong_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
+            out = torch.empty_like(reflectors).to(wrong_device)
+            with self.assertRaisesRegex(RuntimeError, "Expected result and input to be on the same device"):
                 torch.orgqr(reflectors, tau, out=out)
 
-                # device of tau and input should match
-                wrong_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
-                tau = tau.to(wrong_device)
-                # with self.assertRaisesRegex(RuntimeError, "Expected input and tau to be on the same device"):
+            # device of tau and input should match
+            wrong_device = 'cpu' if self.device_type != 'cpu' else 'cuda'
+            tau = tau.to(wrong_device)
+            with self.assertRaisesRegex(RuntimeError, "Expected input and tau to be on the same device"):
                 torch.orgqr(reflectors, tau)
 
+    @onlyCUDA
+    @skipCUDAIfNoMagmaAndNoCusolver
+    @dtypes(torch.float32)
+    def test_orgqr_many_batches(self, device, dtype):
```

Comment:
How long does this test take to run?

Reply:
It looks scary, yes, but it's faster than other tests!
```
3.98s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_complex128
0.40s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_float64
0.33s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_complex128
0.24s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_complex64
0.23s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_float32
0.18s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_errors_and_warnings_cuda
0.14s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_complex64
0.14s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_many_batches_cuda_float32
0.07s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_errors_and_warnings_cpu
0.05s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_float32
0.05s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_float64

```
And when MAGMA is not available (using cuSOLVER then)
```
4.40s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_complex128
0.34s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_float32
0.34s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_float64
0.30s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_complex128
0.29s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_cuda_complex64
0.22s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_errors_and_warnings_cuda
0.19s call     test/test_linalg.py::TestLinalgCUDA::test_orgqr_many_batches_cuda_float32
0.03s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_errors_and_warnings_cpu
0.02s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_complex64
0.01s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_float32
0.01s call     test/test_linalg.py::TestLinalgCPU::test_orgqr_cpu_float64
```